client/utils/utils.go

```diff
@@ -32,12 +32,44 @@ func SendTx(txCtx authctx.TxContext, cliCtx context.CLIContext, msgs []sdk.Msg)
 		return err
 	}
 
+	txCtxPtr, err := lookupTxCtx(txCtx, cliCtx, from)
+	if err != nil {
+		return err
+	}
+	txCtx = *txCtxPtr
+
+	name, err := cliCtx.GetFromName()
+	if err != nil {
+		return err
+	}
+	passphrase, err := keys.GetPassphrase(name)
+	if err != nil {
+		return err
+	}
+
+	if cliCtx.Gas == 0 {
+		txCtx, err = EnrichCtxWithGas(txCtx, cliCtx, name, passphrase, msgs)
+		if err != nil {
+			return err
+		}
+	}
+
+	// build and sign the transaction
+	txBytes, err := txCtx.BuildAndSign(name, passphrase, msgs)
+	if err != nil {
+		return err
+	}
+	// broadcast to a Tendermint node
+	return cliCtx.EnsureBroadcastTx(txBytes)
+}
+
+func lookupTxCtx(txCtx authctx.TxContext, cliCtx context.CLIContext, from []byte) (*authctx.TxContext, error) {
```

Comment:
Where did this come from? I don't think we should be returning pointers to contexts. Is this from the original simulation PR @alessio ?

Reply:
Nvm, I don't see this on `develop`. I don't think `lookupTxCtx` is appropriate here. See `prepareTxContext ` [here](https://github.com/cosmos/cosmos-sdk/blob/181a5162871bf204bbf88d72baee5ac7e8cad9ae/client/utils/utils.go). I think it should be analogous to that.

Reply:
Was trying to reduce the cyclomatic complexity of the function above - I can revert.

Reply:
ohhh no, we should definitely reduce cyclomatic complexity! We just should:
1. Not return a pointer
2. Avoid a merge conflict cause @alessio's PR has something similar

That's all :-)